anatomy/tracks/practice-exercises.md

```diff
@@ -28,4 +28,334 @@ Practice Exercise metadata is defined in the `exercises.practice` key in the [co
+
+He answers 'Calm down, I know what I'm doing!' if you yell a question at him.
+
+He says 'Fine. Be that way!' if you address him without actually saying anything.
+
+He answers 'Whatever.' to anything else.
+```
+
+### File: `.docs/instructions.append.md`
+
+**Purpose:** Additional instructions text to append after the existing instructions.
+
+**Presence:** Optional
+
+In some (rare) cases, you might want to expand on the exercise's `instructions.md` file, for example when the exercise has implemented tests that are not covered by the existing instructions.
+
+```markdown
+Bob's conversational partner is a purist when it comes to written communication and always follows normal rules regarding sentence punctuation in English.
+```
+
+### File: `.docs/hints.md`
+
+**Purpose:** Provide hints to a student to help them get themselves unstuck in an exercise.
+
+**Presence:** Required
+
+- If the student gets stuck, we will allow them to click a button requesting a hint, which will show the relevant part of file.
+- Hints should be bullet-pointed underneath headings.
+- The hints should be enough to unblock almost any student.
+- The hints should not spell out the solution, but instead point to a resource describing the solution (e.g. linking to documentation for the function to use).
+- The hints may use code samples to explain concepts, but not to outline the solution. e.g. in a lists exercise they might show a snippet of how a certain list function works, but not in a way that is directly copy/pasteable into the solution.
+- The hints must appear as a Markdown list under a `## General` heading.
+
+Viewing hints will not be a "recommended" path and we will (softly) discourage using it unless the student can't progress without it. As such, it's worth considering that the student reading it will be a little confused/overwhelmed and maybe frustrated.
+
+#### Example
+
+```markdown
+## General
+
+- There are many [built-in methods](integers) to simplify working with integers.
+
+[integers]: https://ruby-doc.org/core-2.7.0/Integer.html
+```
+
+---
+
+### File: .meta/design.md
+
+**Purpose:** Describe the design of the exercise.
+
+**Presence:** Required
+
+This file contains information on the exercise's design, which includes things like its goal, its teaching goals, what not to teach, and more.
+
+It exists in order to inform future maintainers or contributors about the scope and limitations of an exercise, to avoid the natural trend towards making exercises more complex over time.
+
+#### Example
+
+```markdown
+## Goal
+
+The goal of this exercise is help students practice how to work with strings.
+
+## Notes
+
+This exercise does not contain any error handling tests.
+```
+
+---
+
+### File: `.meta/config.json`
+
+**Purpose:** Contains meta information on the exercise.
+
+**Presence:** Required
+
+This file contains meta information on the exercise:
+
+- `authors`: The exercise's author(s) (required)
```

Comment:
I beg you not to do this for practice exercises 😭 we would need to go through git history for 100 exercises to figure out for each who authored it.

Reply:
Agree. How about we make it optional. I do want it there so people can have that credit, but the work involved to go back through old exercises does feel prohibitive for it being required.

Reply:
> I beg you not to do this for practice exercises 😭 we would need to go through git history for 100 exercises to figure out for each who authored them.

It sounds like I should be able to do this via a script, right?

Reply:
I don't know. How would you do it? Would you choose a single file from an exercise and find the author of the commit that adds that file? Which file would it be? Would it be guaranteed to detect renaming and moving the file around? We changed all the paths in all the exercises once when migrating from Elixir scripts to mix projects. I need to use my IDE's git plugin to correctly track changes to a file, GitHub for example doesn't understand moving files 🙄. And would that really mean that is the real author of the exercise?

Let's take `queen-attack` for example. This is the commit where it was authored by dalexj: https://github.com/exercism/elixir/commit/5648c33704d798001025fe5f82212b6a2c73803f None of the files exists anymore under the same path as they were created. And its README seems to have been created later, by Katrina https://github.com/exercism/elixir/commit/400cfe5a3b9f856b6e752dc348eb1b15cdf9bbf4 When you ask GitHub for the history of the file directory, it only shows the v3 big bang commit: https://github.com/exercism/elixir/commits/fb333fcb94501082b0d174c949d85bc15bfdbc5c/exercises/practice/queen-attack

Reply:
Actually, I think we _must_ do this via a script. There are so many practice exercises implemented and it would be fantastic to attribute them to users.

Reply:
@angelikatyborska That is due to GitHub not using the `--follow` command for the log. See the following console output for the log of the `exercises/practice/wordy/README.md` file in the Elixir repo.

Without --follow:

```
erik@VELPIE:~/elixir$ git log exercises/practice/wordy/README.md
commit fb333fcb94501082b0d174c949d85bc15bfdbc5c
Author: Erik Schierboom <erik_schierboom@hotmail.com>
Date:   Fri Jan 29 14:37:47 2021 +0100

    [v3] Move existing exercises to exercises/practice
```

With --follow:


```
commit fb333fcb94501082b0d174c949d85bc15bfdbc5c
Author: Erik Schierboom <erik_schierboom@hotmail.com>
Date:   Fri Jan 29 14:37:47 2021 +0100

    [v3] Move existing exercises to exercises/practice

commit 7de06dd889c4be1f0bb47f2cadf723ad38e7a201
Author: Cohen Carlisle <cohencarlisle+github@pm.me>
Date:   Sun Jul 28 20:43:09 2019 -0400

    fix old test docs in readmes

commit 432813834b2c09a4a2ed8cf31d36a77b3e3f9ba1
Author: Devon Estes <devon.c.estes@gmail.com>
Date:   Wed Aug 15 11:34:36 2018 +0200

    Run configlet to update READMEs of exercises

....
```